File: app/src/main/java/com/shuyu/apprecycler/chat/detail/dagger/ChatSuperAdapterManager.java
package com.shuyu.apprecycler.chat.detail.dagger;

import android.content.Context;

import com.shuyu.apprecycler.R;
import com.shuyu.apprecycler.bind.view.BindCustomLoadMoreFooter;
import com.shuyu.apprecycler.chat.data.model.ChatImageModel;
import com.shuyu.apprecycler.chat.data.model.ChatTextModel;
import com.shuyu.apprecycler.chat.holder.ChatImageHolder;
import com.shuyu.apprecycler.chat.holder.ChatTextHolder;
import com.shuyu.bind.BindSuperAdapterManager;
import com.shuyu.bind.listener.OnBindDataChooseListener;

import java.util.List;

import javax.inject.Inject;

/**
 * 继承之后实现注入与初始化
 * Created by guoshuyu on 2017/9/5.
 */

@ChatDetailSingleton
public class ChatSuperAdapterManager extends BindSuperAdapterManager {

    @Inject
    public ChatSuperAdapterManager() {
        super();
    }

    @Inject
    public void initManager(Context context) {
        bind(ChatImageModel.class, R.layout.chat_layout_image_left, ChatImageHolder.class)
                .bind(ChatImageModel.class, R.layout.chat_layout_image_right, ChatImageHolder.class)
                .bind(ChatTextModel.class, R.layout.chat_layout_text_left, ChatTextHolder.class)
                .bind(ChatTextModel.class, R.layout.chat_layout_text_right, ChatTextHolder.class)
                .bingChooseListener(new OnBindDataChooseListener() {
                    @Override
                    public int getCurrentDataLayoutId(Object object, Class classType, int position, List<Integer> ids) {
                        if (object instanceof ChatTextModel) {
                            ChatTextModel chatTextModel = (ChatTextModel) object;
                            return (chatTextModel.isMe()) ? R.layout.chat_layout_text_right : R.layout.chat_layout_text_left;
                        } else if (object instanceof ChatImageModel) {
                            ChatImageModel chatImageModel = (ChatImageModel) object;
                            return (chatImageModel.isMe()) ? R.layout.chat_layout_image_right : R.layout.chat_layout_image_left;
                        }
                        return ids.get(ids.size() - 1);
                    }
                })
                .setPullRefreshEnabled(false)
                .setLoadingMoreEnabled(false)
                .setFootView(new BindCustomLoadMoreFooter(context))
                .setNeedAnimation(true);
    }

}

File: app-kotlin/src/main/java/example/shuyu/recycler/kotlin/chat/detail/dagger/ChatSuperAdapterManager.kt
package example.shuyu.recycler.kotlin.chat.detail.dagger

import android.content.Context
import android.view.View

import com.shuyu.commonrecycler.BindSuperAdapterManager
import com.shuyu.commonrecycler.listener.OnBindDataChooseListener
import com.shuyu.commonrecycler.listener.OnItemClickListener
import com.shuyu.commonrecycler.listener.OnLoadingListener

import javax.inject.Inject

import example.shuyu.recycler.kotlin.R
import example.shuyu.recycler.kotlin.bind.view.BindCustomLoadMoreFooter
import example.shuyu.recycler.kotlin.chat.data.model.ChatImageModel
import example.shuyu.recycler.kotlin.chat.data.model.ChatTextModel
import example.shuyu.recycler.kotlin.chat.holder.ChatImageHolder
import example.shuyu.recycler.kotlin.chat.holder.ChatTextHolder

/**
 * 继承之后实现注入与初始化
 * Created by guoshuyu on 2017/9/5.
 */

@ChatDetailSingleton
open class ChatSuperAdapterManager @Inject constructor() : BindSuperAdapterManager() {

    @Inject
    fun initManager(context: Context, onItemClickListener: OnItemClickListener, loadingListener: OnLoadingListener, onTouchListener: View.OnTouchListener) {
        bind(ChatImageModel::class.java, R.layout.chat_layout_image_left, ChatImageHolder::class.java)
                .bind(ChatImageModel::class.java, R.layout.chat_layout_image_right, ChatImageHolder::class.java)
                .bind(ChatTextModel::class.java, R.layout.chat_layout_text_left, ChatTextHolder::class.java)
                .bind(ChatTextModel::class.java, R.layout.chat_layout_text_right, ChatTextHolder::class.java)
                .bingChooseListener(object : OnBindDataChooseListener {
                    override fun getCurrentDataLayoutId(`object`: Any, classType: Class<*>, position: Int, ids: List<Int>): Int {
                        if (`object` is ChatTextModel) {
                            return if (`object`.isMe) R.layout.chat_layout_text_right else R.layout.chat_layout_text_left
                        } else if (`object` is ChatImageModel) {
                            return if (`object`.isMe) R.layout.chat_layout_image_right else R.layout.chat_layout_image_left
                        }
                        return ids[ids.size - 1]
                    }
                })
                .setPullRefreshEnabled(false)
                .setLoadingMoreEnabled(false)
                .setFootView(BindCustomLoadMoreFooter(context))
                .setNeedAnimation(true)
                .setTouchListener(onTouchListener)
                .setOnItemClickListener(onItemClickListener)
                .setLoadingListener(loadingListener)
    }

}
